src/panels/KaitoModelsPanel.ts

```diff
@@ -0,0 +1,295 @@
+    }
+}
+// interface DeploymentProgressUpdate {
+//     modelName: string;
+//     workspaceExists: boolean;
+//     resourceReady: boolean | null;
+//     inferenceReady: boolean | null;
+//     workspaceReady: boolean | null;
+//     age: number | null;
+// }
+export class KaitoModelsPanelDataProvider implements PanelDataProvider<"kaitoModels"> {
+    public constructor(
+        readonly clusterName: string,
+        readonly subscriptionId: string,
+        readonly resourceGroupName: string,
+        readonly armId: string,
+        readonly kubectl: k8s.APIAvailable<k8s.KubectlV1>,
+        readonly kubeConfigFilePath: string,
+    ) {
+        this.clusterName = clusterName;
+        this.subscriptionId = subscriptionId;
+        this.resourceGroupName = resourceGroupName;
+        this.armId = armId;
+        this.kubectl = kubectl;
+        this.kubeConfigFilePath = kubeConfigFilePath;
+    }
+    private cancelToken: boolean = false;
+    getTitle(): string {
+        return `Create KAITO Workspace`;
+    }
+    getInitialState(): InitialState {
+        return {
+            clusterName: this.clusterName,
+            modelName: "",
+            workspaceExists: false,
+            resourceReady: false,
+            inferenceReady: false,
+            workspaceReady: false,
+            age: 0,
+        };
+    }
+    cancel() {
+        this.cancelToken = true;
+        // vscode.window.showInformationMessage("Operation cancelled"); - debug 1
+    }
+    getTelemetryDefinition(): TelemetryDefinition<"kaitoModels"> {
+        return {
+            generateCRDRequest: true,
+            deployKaitoRequest: true,
+            workspaceExistsRequest: false,
+            updateStateRequest: false,
+            resetStateRequest: false,
+            cancelRequest: false,
+        };
+    }
+    getMessageHandler(webview: MessageSink<ToWebViewMsgDef>): MessageHandler<ToVsCodeMsgDef> {
+        return {
+            generateCRDRequest: (params) => {
+                this.handleGenerateCRDRequest(params.model);
+            },
+            deployKaitoRequest: (params) => {
+                this.handleDeployKaitoRequest(params.model, params.yaml, webview);
+            },
+            workspaceExistsRequest: (params) => {
+                this.handleWorkspaceExistsRequest(params.model);
+            },
+            updateStateRequest: (params) => {
+                this.handleUpdateStateRequest(params.model, webview);
+            },
+            resetStateRequest: () => {
+                this.handleResetStateRequest(webview);
+            },
+            cancelRequest: () => {
+                this.cancel();
+            },
+        };
+    }
+
+    private async handleGenerateCRDRequest(yaml: string) {
+        const doc = await vscode.workspace.openTextDocument({
+            content: yaml,
+            language: "yaml",
+        });
+        vscode.window.showTextDocument(doc);
+    }
+
+    nullIsFalse(value: boolean | null): boolean {
```

Comment:
💡 How about this:

This below just uses the `nullish` coalescing operator `(??)`, which returns the right-hand operand (false) if the left-hand operand (value) is null or undefined. Otherwise, it returns value.

```*.ts
nullIsFalse(value: boolean | null): boolean {
    return value ?? false;
}
```